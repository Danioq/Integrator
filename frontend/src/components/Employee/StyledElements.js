import styled from 'styled-components';

export let Card = styled.li`
  display: inline-block;
  align-items: center;
  margin: 10px;
  min-height: 30vh;
  width: 11vw;
  padding: 10px;
  text-align: center;
  border: 1px solid #03e008;
  border-radius: 15px;
  box-shadow: 1px 1px 1px 1px #03b008;
  background-color: rgb(225, 242, 225);
  transition: all 0.5s;
  &:hover {
    transform: scale(1.1);
    box-shadow: none;
  }
  @media (max-width: 600px) {
    width: 50vw;
  }
`;

export let Picture = styled.img`
  border-radius: 15px;
`;

export let Name = styled.p`
  font-weight: bolder;
  font-size: 1.1em;
  cursor: default;
`;

export let Age = styled.p`
cursor: default;
`;

export let District = styled.p`
cursor: default;
`;

export let Department = styled.p`
cursor: default;
`;