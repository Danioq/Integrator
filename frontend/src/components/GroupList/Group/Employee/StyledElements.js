import styled from 'styled-components';

export let Card = styled.div`

`;

export let Picture = styled.img``;

export let Name = styled.p``;

export let Age = styled.p``;

export let District = styled.p``;

export let Department = styled.p``;