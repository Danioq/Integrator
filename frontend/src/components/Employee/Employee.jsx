import React from 'react';
import {Card, Picture, Name, Age, District, Department} from './StyledElements'
import PropTypes from 'prop-types';

function Employee(props){
  return<Card>
    <Picture src={props.employee.picture}/>
    <Name>{`${props.employee.first_name} ${props.employee.last_name}`}</Name>
    <Age>Wiek<br/>{props.employee.age}</Age>
    <District>Dzielnica<br/> {props.employee.district}</District>
    <Department>Dzia≈Ç <br/> {props.employee.department}</Department>
  </Card>;
}

Employee.propTypes = {
  employee : PropTypes.shape({
    picture: PropTypes.string.isRequired,
    first_name: PropTypes.string.isRequired,
    last_name: PropTypes.string.isRequired,
    age: PropTypes.number.isRequired,
    district: PropTypes.string.isRequired,
    department: PropTypes.string.isRequired
  })
};
export default Employee;