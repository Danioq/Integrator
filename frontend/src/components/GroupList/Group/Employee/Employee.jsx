import React from 'react';
import {Card, Picture, Name, Age, District, Department} from './StyledElements'

export default function Employee(props){
  return<Card>
    <Picture src={props.picutre}/>
    <Name>{props.name}</Name>
    <Age>{props.age}</Age>
    <District>{props.district}</District>
    <Department>{props.department}</Department>
  </Card>;
}