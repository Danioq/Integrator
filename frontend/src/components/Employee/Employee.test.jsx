import Employee from './Employee';
import renderer from 'react-test-renderer';
import React from 'react';

describe('Employee component suite', () => {
  test('Render test', () => {
    let empData = {
      picture: "pic.png",
      first_name: "xyz",
      last_name: "nowak",
      age: 20,
      district: "krzyki",
      department: "IT"
    }
    let component = renderer.create(<Employee employee={empData} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});