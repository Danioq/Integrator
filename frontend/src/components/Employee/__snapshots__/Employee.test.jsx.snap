// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Employee component suite Render test 1`] = `
<li
  className="sc-AxjAm bltkeM"
>
  <img
    className="sc-AxirZ oOxgC"
    src="pic.png"
  />
  <p
    className="sc-AxiKw bmJKWB"
  >
    xyz nowak
  </p>
  <p
    className="sc-AxhCb jKUKVm"
  >
    Wiek
    <br />
    20
  </p>
  <p
    className="sc-AxhUy cQPCGv"
  >
    Dzielnica
    <br />
     
    krzyki
  </p>
  <p
    className="sc-AxgMl lhRGCN"
  >
    Dzia≈Ç 
    <br />
     
    IT
  </p>
</li>
`;
